eZ/Publish/Core/MVC/Symfony/Controller/Content/ViewController.php

```diff
@@ -102,39 +102,14 @@ protected function buildResponse($etag = null, DateTime $lastModified = null)
      */
     public function viewLocation($locationId, $viewType, $layout = false, array $params = array())
     {
-        $this->performAccessChecks();
-        $response = $this->buildResponse();
-
-        try {
-            if (isset($params['location']) && $params['location'] instanceof Location) {
-                $location = $params['location'];
-            } else {
-                $location = $this->getRepository()->getLocationService()->loadLocation($locationId);
-                if ($location->invisible) {
-                    throw new NotFoundHttpException("Location #$locationId cannot be displayed as it is flagged as invisible.");
-                }
+        if (!isset($params['location'])) {
+            $params['location'] = $this->getRepository()->getLocationService()->loadLocation($locationId);
+            if ($params['location']->invisible) {
+                throw new NotFoundHttpException("Location #$locationId cannot be displayed as it is flagged as invisible.");
             }
-
-            $response->headers->set('X-Location-Id', $locationId);
-            $response->setContent(
-                $this->renderLocation(
-                    $location,
-                    $viewType,
-                    $layout,
-                    $params
-                )
-            );
-
-            return $response;
-        } catch (UnauthorizedException $e) {
-            throw new AccessDeniedException();
-        } catch (NotFoundException $e) {
-            throw new NotFoundHttpException($e->getMessage(), $e);
-        } catch (NotFoundHttpException $e) {
-            throw $e;
-        } catch (Exception $e) {
-            return $this->handleViewException($response, $params, $e, $viewType, null, $locationId);
         }
+
+        return $this->viewContent($params['location']->contentId, $viewType, $layout, $params);
```

Comment:
Wouldn't it make sense to use `$this->forward` instead? It's a bit of overhead, but we could then take advantage of param converters